components/builder-jobsrv/src/scheduler_datastore/test.rs

```diff
@@ -0,0 +1,358 @@
+#[cfg(test)]
+#[cfg(feature = "postgres_tests")]
+// cargo test --features postgres_tests to enable
+// from root
+// cargo test -p habitat_builder_jobsrv --features=postgres_tests
+// --manifest-path=components/builder-jobsrv/Cargo.toml
+mod test {
+    use crate::{assert_match,
+                data_store::DataStore,
+                hab_core::package::PackageTarget,
+                test_helpers::*};
+
+    use habitat_builder_db::{datastore_test,
+                             models::{jobs::{JobExecState,
+                                             JobGraphEntry,
+                                             JobStateCounts,
+                                             NewJobGraphEntry},
+                                      package::BuilderPackageTarget}};
+    use std::str::FromStr;
+
+    mod helpers {
+
+        pub fn manifest_data_from_file() -> Vec<(String, String, Vec<String>)> {
+            let manifest = include_str!("manifest_data.txt");
+            let mut data = Vec::new();
+            for line in manifest.lines() {
+                let fields: Vec<String> = line.split_whitespace().map(|x| x.to_string()).collect();
+                let deps: Vec<String> = fields[2].split(',').map(|x| x.to_string()).collect();
+                data.push((fields[0].clone(), fields[1].clone(), deps));
+            }
+            data
+        }
+    }
+
+    #[test]
+    fn create_job_graph_entry() {
+        let target_platform =
+            BuilderPackageTarget(PackageTarget::from_str("x86_64-linux").unwrap());
+        let ds = datastore_test!(DataStore);
+        let conn = ds.get_pool().get_conn().unwrap();
+        let entry = NewJobGraphEntry { group_id: 0,
+                                       job_state: JobExecState::Pending,
+                                       project_name: "foo/bar",
+                                       job_id: None,
+                                       manifest_ident: "foo/bar/1.2.3/123",
+                                       as_built_ident: None,
+                                       dependencies: &[1, 2, 3],
+                                       waiting_on_count: 3,
+                                       target_platform };
+
+        let job_graph_entry = JobGraphEntry::create(&entry, &conn).unwrap();
+
+        assert_eq!(job_graph_entry.group_id, 0);
+        assert_eq!(job_graph_entry.job_state, JobExecState::Pending);
+    }
+
+    #[test]
+    fn take_next_job_for_target() {
+        let target_platform =
+            BuilderPackageTarget(PackageTarget::from_str("x86_64-linux").unwrap());
```

Comment:
This might benefit from a `Default` implementation on `BuilderPackageTarget`, or (failing that), a small helper function to simplify the construction of it.